sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdk.java

```diff
@@ -51,11 +53,13 @@ public Scope withSpan(Span span) {
   }
 
   @Override
-  public void setResource(Resource resource) {}
+  public void setResource(Resource resource) {
+    this.resource = resource;
+  }
 
   @Override
   public Resource getResource() {
```

Comment:
is there a comment on API to not cache this value? We need to clearly state that the object returned by this API can be replaced in runtime at any moment

Reply:
Some questions:
1. I am not sure why do we need a get on this?
1. Also not sure we need a set, I think we can just offer the ability to set the initial resource during initialization time of the TracerSdk.

Reply:
I did this PR to move forward with the impl. Setter and getter on the resource seem not required in my opinion. I thought that this has been discussed as it is in the API.


I think that setting a resource belongs to initialization. Getter seems also redundant.

Reply:
@pavolloffay I am sorry that I kind of moved the discussion a bit here and that it happen in your PR. You did a good job on this PR, but want to make sure we do the best of the project.

I think we are on the same opinion about what is needed here.

Reply:
So do we remove these methods completely fro the tracer API?

Reply:
That's what I am proposing, waiting for @SergeyKanzhelev and @carlosalberto to see what do they think?

Reply:
Oh, I remember we talking about this some time ago ;) Yeah, it makes sense to have this as part of the `Tracer` initialization only - leaving the `get()` part would be fine, but probably we don't **really** need to expose it in the main API.

Reply:
One reason to expose it is to allow to inject this resource information in outgoing requests and response headers. There are systems that exchange this information.

Reply:
Yeah, I'd propose go with the 'safe' route now and remove it for the moment, and add it if/when actually needed.